qa/zcash/smoke_tests.py

```diff
@@ -0,0 +1,800 @@
+            results[case] = False
+        return None
+
+def wait_for_balance(zcash, zaddr, expected=None, timeout=900):
+    print('Waiting for funds to %s...' % zaddr)
+    if expected is not None:
+        print('Expecting %s' % expected)
+
+    unconfirmed_balance = zcash.z_getbalance(zaddr, 0)
+    print('Unconfirmed balance: %s' % unconfirmed_balance)
+    if expected is not None and unconfirmed_balance != expected:
+        print('WARNING: Unconfirmed balance does not match expected balance')
+
+    ttl = timeout
+    while True:
+        balance = zcash.z_getbalance(zaddr)
+        if balance == unconfirmed_balance:
+            print('Received %s' % balance)
+            return balance
+
+        time.sleep(1)
+        ttl -= 1
+        if ttl == 0:
+            # Ask user if they want to keep waiting
+            print()
+            ret = input('Do you wish to continue waiting? (Y/n) ')
+            if ret.to_lower() == 'n':
+                print('Address contained %s at timeout' % balance)
+                return balance
+            else:
+                ttl = 300
+
+def wait_and_check_balance(results, case, zcash, addr, expected):
+    balance = wait_for_balance(zcash, addr, expected)
+    if balance != expected and results is not None and len(case) > 0:
+        results[case] = False
+    return balance
+
+def wait_for_txid_operation(zcash, opid, timeout=300):
+    print('Waiting for async operation %s' % opid)
+    result = None
+    for _ in iter(range(timeout)):
+        results = zcash.z_getoperationresult([opid])
+        if len(results) > 0:
+            result = results[0]
+            break
+        time.sleep(1)
+
+    status = result['status']
+    if status == 'failed':
+        print('Operation failed')
+        print(result['error']['message'])
+        return None
+    elif status == 'success':
+        txid = result['result']['txid']
+        print(txid)
+        return txid
+
+def async_txid_cmd(results, case, zcash, name, args=[]):
+    opid = run_cmd(results, case, zcash, name, args)
+    # Some async commands return a dictionary containing the opid
+    if type(opid) == type({}):
+        opid = opid['opid']
+    if opid is None:
+        if results is not None and len(case) > 0:
+            results[case] = False
+        return None
+
+    txid = wait_for_txid_operation(zcash, opid)
+    if txid is None:
+        if results is not None and len(case) > 0:
+            results[case] = False
+    return txid
+
+def z_sendmany(results, case, zcash, from_addr, recipients):
+    return async_txid_cmd(results, case, zcash, 'z_sendmany', [
+        from_addr,
+        [{
+            'address': to_addr,
+            'amount': amount,
+        } for (to_addr, amount) in recipients]
+    ])
+
+def check_z_sendmany(results, case, zcash, from_addr, recipients):
+    txid = z_sendmany(results, case, zcash, from_addr, recipients)
+    if txid is None:
+        return decimal.Decimal('0')
```

Comment:
We should return a list here.

Not returning a list causes the following to throw an error:
```
taddr_balance = check_z_sendmany(
    results, '4c', zcash, sprout_zaddr_1, [(taddr_1, (starting_balance / 10) * 6)])[0]
```

Reply:
I think there are places where we are returning `None` that can cause similar errors too.